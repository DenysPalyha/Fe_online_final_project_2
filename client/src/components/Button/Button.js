// eslint-disable-next-line no-unused-vars
import React, { memo } from 'react';
import PropTypes from 'prop-types';

function Button(props) {
  const {
    children, onClick, ...rest
  } = props;

  return (
  // eslint-disable-next-line react/jsx-props-no-spreading
    <button type="button" onClick={onClick} {...rest}>
      {children}
    </button>
  );
}

Button.propTypes = {
  onClick: PropTypes.func,
  type: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

Button.defaultProps = {
  onClick: () => {},
  type: 'button',
  children: null,

};

export default memo(Button);
