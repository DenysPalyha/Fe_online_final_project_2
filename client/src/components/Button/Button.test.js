import { render } from '@testing-library/react';
import Button from './Button';
// eslint-disable-next-line no-undef
describe('Button snapshot testing', () => {
  // eslint-disable-next-line no-undef
  test('should Button match snapshot', () => {
    const { asFragment } = render(<Button />);
    // eslint-disable-next-line no-undef
    expect(asFragment()).toMatchSnapshot();
  });
});
