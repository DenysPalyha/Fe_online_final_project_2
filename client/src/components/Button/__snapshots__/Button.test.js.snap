// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot testing should Button match snapshot 1`] = `
<DocumentFragment>
  <button
    type="button"
  />
</DocumentFragment>
`;
